cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -36,6 +38,20 @@ func appendTargetError(res *worker.OSBuildJobResult, err error) {
 	res.TargetErrors = append(res.TargetErrors, errStr)
 }
 
+// Returns an *awsupload.AWS object with the credentials of the request. If they are not accessible, then try to use the
+// one obtained in the worker configuration
+func (impl *OSBuildJobImpl) getAWS(region string, accessId string, secret string, token string) (*awsupload.AWS, error) {
+	if accessId != "" && secret != "" {
+		return awsupload.New(region, accessId, secret, token)
+	} else {
+		cred, err := impl.AWSCreds.Get()
+		if err != nil {
+			return nil, err
+		}
+		return awsupload.New(region, cred.AccessKeyID, cred.SecretAccessKey, cred.SessionToken)
+	}
```

Comment:
We should also support a third option (which we should actually be using in our workers): no credentials are passed in at all, and the AWS SDK will use the instance credentials, in case the VM is running in EC2. See: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials

We may need to change the API of `awsupload.New()` to take a `creds` object as argument, rather than the individual pieces. And the fallback would then be to pass `nil`, which should (though I did not try) use instance credentials.

Reply:
I agree.
Well I think, in the end, we can have two "constructors" for our awsupload.AWS object, one that initialize it with individual pieces, and another that initializes it with a filename.